src/coreclr/src/vm/assemblynative.cpp

```diff
@@ -238,9 +238,10 @@ void QCALLTYPE AssemblyNative::LoadFromPath(INT_PTR ptrNativeAssemblyLoadContext
 
     if (pwzILPath != NULL)
     {
+
         pILImage = PEImage::OpenImage(pwzILPath,
-                                      MDInternalImport_Default,
-                                      Bundle::ProbeAppBundle(pwzILPath));
+                                      MDInternalImport_NoCache,
```

Comment:
We need to make sure that code like this keeps working:
```C#
            Assembly pluginAssembly = testAlc.LoadFromAssemblyPath(filePath);
            Assembly pluginAssemblySecondLoad = testAlc.LoadFromAssemblyPath(filePath);
            Assert.Same(pluginAssembly, pluginAssemblySecondLoad);
```

Meaning that trying to load the same file to the same ALC multiple times returns the same assembly.
It's likely that the binding code below will realize that it's the same assembly and return the existing loaded assembly. That said, this could still be a perf problem since we're likely going to map the file and parse headers more than we did before.

I think the better fix in this case should be something in the PEImage cache, were we make it bundle aware. Basically make sure that the key is not just path but (path, isInBundle) instead. Looking through the code such change doesn't seem too complicated.

Reply:
I agree that the cache taking into account the bundle is the right thing here.

`MDInternalImport_NoCache` means both don't look in the cache and don't cache the results. I'd think we still want to use/update the cache in both normal and single-file.